File: 0011-container-with-most-water/0011-container-with-most-water.java
class Solution {
    public int maxArea(int[] height) {

        int maxi = 0;
        int left = 0;
        int right = height.length -1;

        while(left < right){
            int area  = Math.min(height[left], height[right]) * (right - left);

            maxi = Math.max(maxi, area);

            if(height[left] < height[right]){
                left++;
            }else{
                right--;
            }
        }

        return maxi;
        
    }
}

File: 0011-container-with-most-water/0011-container-with-most-water.js
var maxArea = function (height) {
  let left = 0;
  let right = height.length - 1;

  let max = 0;

  while (left < right) {
    let curr = Math.min(height[left], height[right]) * (right - left);
    if (curr > max) max = curr;

    if (height[left] < height[right]) {
      left++;
    } else {
      right--;
    }
  }

  return max;
};
